openapi/provider_test.go

```diff
@@ -193,6 +195,134 @@ definitions:
+		}
+		xTerraformResourceName2 := ""
+		if preferredName2 != "" {
+			xTerraformResourceName2 = `x-terraform-resource-name: "` + preferredName2 + `"`
+		}
+		swagger := `swagger: "2.0"
+paths:
+  ` + path1 + `:
+    post:
+      ` + xTerraformResourceName1 + `
+      parameters:
+      - in: "body"
+        name: "body"
+        schema:
+          $ref: "#/definitions/whatever"
+      responses:
+        201:
+          schema:
+            $ref: "#/definitions/whatever"
+  ` + path1 + `/{id}:
+    get:
+      parameters:
+      - name: "id"
+        in: "path"
+        type: "string"
+      responses:
+        200:
+          schema:
+            $ref: "#/definitions/whatever"
+
+  ` + path2 + `:
+    post:
+      ` + xTerraformResourceName2 + `
+      parameters:
+      - in: "body"
+        name: "body"
+        schema:
+          $ref: "#/definitions/whatever"
+      responses:
+        201:
+          schema:
+            $ref: "#/definitions/whatever"
+  ` + path2 + `/{id}:
+    get:
+      parameters:
+      - name: "id"
+        in: "path"
+        type: "string"
+      responses:
+        200:
+          schema:
+            $ref: "#/definitions/whatever"
+
+definitions:
+  whatever:
+    type: "object"
+    properties:
+      id:
+        type: "string"
+        readOnly: true`
+		return swagger
+	}
+
+	Convey("Given a swagger doc that declares resources with colliding names, "+
+		"When CreateSchemaProviderWithConfiguration method is called, "+
+		"Then there should be no error but the provider should not have those resources and a warning should be logged", t, func() {
+
+		testcases := []struct {
+			label           string
+			path1           string
+			preferredName1  string
+			path2           string
+			preferredName2  string
+			expectedWarning string
+		}{
+			{label: "resources with colliding x-terraform-resource-names",
+				preferredName1:  "collision",
+				preferredName2:  "collision",
+				expectedWarning: "'collision' is a duplicate resource name and is being removed from the provider"},
+			{label: "resources with colliding x-terraform-resource-name calculated name and calculated non-versioned name",
+				path1:           "/collision",
+				preferredName2:  "collision",
+				expectedWarning: "'collision' is a duplicate resource name and is being removed from the provider"},
+			{label: "resources with colliding x-terraform-resource-name calculated name and calculated versioned name",
+				path1:           "/v1/collision",
+				preferredName2:  "collision_v1",
+				expectedWarning: "'collision_v1' is a duplicate resource name and is being removed from the provider"},
+			{label: "resources with colliding calculated names",
+				path1:           "/v1/collision",
+				path2:           "/collision_v1",
+				expectedWarning: "'collision_v1' is a duplicate resource name and is being removed from the provider"},
+		}
+
+		for _, tc := range testcases {
+			f, _ := ioutil.TempFile("", "*")
```

Comment:
if we are using tmp file for logging , let's be sure the clean after ourself with a defer remove tmp file, otherwise can we just use a NewBuffer([]bytes{}) in log.SetOutput() ?

Reply:
Yeah probably.